File: _exercises/archive/02_word-game.py
# %%
import chatlas
import dotenv

dotenv.load_dotenv()

# %% [markdown]
# ## Ley's play a word game!
#
# Set up a `chat` with a system prompt:
#
# > You are playing a word guessing game. At each turn, guess the word and tell
# > us what it is.

# %%
chat = chatlas.____(____="")

# %%
# Ask the first question:
____("In British English, guess the word for a person who lives next door.")

# %%
# Ask the second question:
____("What helps a car move smoothly down the road?")

# %% [markdown]
# Crate a new, empty chat and ask the second question again, by itself.
#
# How do the responses differ? Why?

# %%
chat2 = chatlas.____()
chat2.____("What helps a car move smoothly down the road?")

File: _exercises/02_word-game/02_word-game.R
library(ellmer)

# Let's play a word game!

# 1. Fill in the blanks below to create a `chat` with the following system prompt:
#
# > You are playing a word guessing game. At each turn, guess the word and tell
#   us what it is.
chat <- chat______(
  ____ = ""
)

# 2. Fill in the blank to ask the the first question:
____("In British English, guess the word for the person who lives next door.")

# 3. 2. Fill in the blank to ask the the second question:
_____("What helps a car move smoothly down the road?")

# 4. Create a new chat with no system prompt and ask the second question again.
chat2 <- chat_____()
chat2$____

# 5. How do the answers to 3 and 4 differ? Why?